v4.0.5 -- 1.8.3 Support (Bukkit, Spigot), Language File

meyerzinn opened the pull request:
This is version 5.0.0 for you. It works (tested) on 1.8.X, and most (if not all) 1.7.X builds. It is also tested against Spigot for 1.8.X and 1.7.X.

meyerzinn committed:
Updated to 1.8

Updated a long-lost friend (abandoned project) to 1.8 for anyone to use.

meyerzinn committed:
I think it's time...

Jumping several versions all the way to 1.8 definitely calls for a new
version ^^

meyerzinn committed:
Added Prebuild Jar

SuperSpyTX commented:
I thought I go ahead and say thanks for taking interest in reviving one of my old projects.  While some of the commits you made (which I'll make notes to shortly) raises a few questions, there's only one thing you should change:

Revert the most recent commit "Because MaxMind vandalism is fun" or whatever, since I was pissed that day when MaxMind decided to block my IP for downloading their database directly.

Otherwise there are a couple things that do need to be paid attention to, and I'll go make these notes in these commits shortly.

SuperSpyTX commented:
As for your message about taking over the project, if you can successfully do the above, I'll let you be the maintainer.

meyerzinn commented:
Ern... I'll do it as soon as I get home tonight ^^

meyerzinn commented:
I am (possibly only temp) working on a Mac, so there's your Unix thighs. And I will look at the changes you want, but as it is it works just fine on 1.8.3 and 1.7.10 both Bukkit and Spigot, but I see the code optimization you are suggesting.

meyerzinn commented:
Thighs? Derping, Perms

SuperSpyTX commented:
I think it may be easy enough to do something like chmod -R 0644 src/* in the terminal.

EDIT: It should work normally on both Bukkit & Spigot 1.8/1.7.10 with the Metrics change, which explains why AntiBot's metrics have shown that the plugin has been deserted.

meyerzinn commented:
I will change the perms asap, but why exactly do they matter? (Never been a problem for me :3)

Sent from my iPad

> On Apr 1, 2015, at 11:48 AM, Joe notifications@github.com wrote:
> 
> I think it may be easy enough to do something like chmod -R 0644 src/\* in the terminal.
> 
> EDIT: It should work normally on both Bukkit & Spigot 1.8/1.7.10 with the Metrics change, which explains why AntiBot's metrics have shown that the plugin has been deserted.
> 
> —
> Reply to this email directly or view it on GitHub.

meyerzinn committed:
Language File!

I added a real language file (using
https://bukkit.org/threads/language-files.149837/, to be fair). Version
4.0.5. Language file might not work everywhere. Also, I need to make a
page on BukkitDev explaining what variables can be used where.

meyerzinn commented:
I just added the Language file.

> On Apr 1, 2015, at 11:48 AM, Joe notifications@github.com wrote:
> 
> I think it may be easy enough to do something like chmod -R 0644 src/\* in the terminal.
> 
> EDIT: It should work normally on both Bukkit & Spigot 1.8/1.7.10 with the Metrics change, which explains why AntiBot's metrics have shown that the plugin has been deserted.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/SuperSpyTX/AntiBot/pull/13#issuecomment-88538129.

SuperSpyTX commented:
The file permissions stuff is so there isn't any messy permissions already being applied.  755 I think asserts that the file can be accessed by anybody in the file system, and I'm sure while the idea of that isn't entirely an issue with source code, I rather keep it platform friendly.

meyerzinn commented:
Alright. I'll mess with the permissions later.

Sent from my iPad

> On Apr 1, 2015, at 8:01 PM, Joe notifications@github.com wrote:
> 
> The file permissions stuff is so there isn't any messy permissions already being applied. 755 I think asserts that the file can be accessed by anybody in the file system, and I'm sure while the idea of that isn't entirely an issue with source code, I rather keep it platform friendly.
> 
> —
> Reply to this email directly or view it on GitHub.

SuperSpyTX commented:
I think you partially fixed it since I see commits that were going from 0755 -> 0644.

Either way, don't worry too much about it.

SuperSpyTX commented:
Do you use Steam by any chance?

meyerzinn commented:
I tried :P it messed up my IDE and GitHub for Mac, so I had to revert some.

Sent from my iPad

> On Apr 1, 2015, at 8:11 PM, Joe notifications@github.com wrote:
> 
> I think you partially fixed it since I see commits that were going from 0755 -> 0644.
> 
> Either way, don't worry too much about it.
> 
> —
> Reply to this email directly or view it on GitHub.

meyerzinn commented:
I have it. I'm not extremely active on it anymore. I'd be 20zinnm.

Sent from my iPad

> On Apr 1, 2015, at 8:11 PM, Joe notifications@github.com wrote:
> 
> Do you use Steam by any chance?
> 
> —
> Reply to this email directly or view it on GitHub.

meyerzinn commented:
Do you by any chance code in Objective C or C++?

Sent from my iPad

> On Apr 1, 2015, at 8:11 PM, Joe notifications@github.com wrote:
> 
> Do you use Steam by any chance?
> 
> —
> Reply to this email directly or view it on GitHub.

meyerzinn commented:
And if we decide to keep the Language file separate for ease of swapping, feel free to add me to the project so I can make a page for it.

Sent from my iPad

> On Apr 1, 2015, at 8:11 PM, Joe notifications@github.com wrote:
> 
> Do you use Steam by any chance?
> 
> —
> Reply to this email directly or view it on GitHub.

SuperSpyTX commented:
I worked with C++ for a brief while, but I have a feeling I'll be working with it a lot more this year.

What's up?

meyerzinn commented:
I am terrible with it, but when you mentioned steam I thought you might be someone I knew :P but I doubt it xD

Sent from my iPad

> On Apr 1, 2015, at 8:16 PM, Joe notifications@github.com wrote:
> 
> I worked with C++ for a brief while, but I have a feeling I'll be working with it a lot more this year.
> 
> What's up?
> 
> —
> Reply to this email directly or view it on GitHub.
